From 631334da464f8eada6606f987a14eb4a4a66224b Mon Sep 17 00:00:00 2001
From: Dan Walkes <danwalkes@trellis-logic.com>
Date: Wed, 31 Oct 2018 12:25:48 -0600
Subject: [PATCH] Jetson TX2 mender boot commands

Modify u-boot to use bootcmd setting based on
values which support jetson-tx2 boot partition layout
and setup.
---
 include/env_default.h | 4 ++--
 include/env_mender.h  | 9 ++++++---
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/include/env_default.h b/include/env_default.h
index 700aedb..ae631c0 100644
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -34,8 +34,8 @@ const uchar default_environment[] = {
 #ifdef	CONFIG_BOOTARGS
 	"bootargs="	CONFIG_BOOTARGS			"\0"
 #endif
-#ifdef	CONFIG_BOOTCOMMAND
-	"bootcmd="	CONFIG_BOOTCOMMAND		"\0"
+#ifdef	CONFIG_MENDER_BOOTCOMMAND
+	"bootcmd="	CONFIG_MENDER_BOOTCOMMAND	"\0"
 #endif
 #ifdef	CONFIG_RAMBOOTCOMMAND
 	"ramboot="	CONFIG_RAMBOOTCOMMAND		"\0"
diff --git a/include/env_mender.h b/include/env_mender.h
index cf46ef0..ac53536 100644
--- a/include/env_mender.h
+++ b/include/env_mender.h
@@ -136,9 +136,12 @@
 
 #define CONFIG_MENDER_BOOTCOMMAND                                       \
     "run mender_setup; "                                                \
-    MENDER_BOOTARGS                                                     \
-    MENDER_LOAD_KERNEL_AND_FDT                                          \
-    "${mender_boot_kernel_type} ${loadaddr} - ${fdt_addr_r}; "     \
+    "setenv distro_bootpart ${mender_boot_part}; "			\
+    "setenv distro_bootpart_hex ${mender_boot_part_hex}; "		\
+    "setenv devnum " __stringify(MENDER_UBOOT_STORAGE_DEVICE) "; " 	\
+    "setenv devtype " __stringify(MENDER_UBOOT_STORAGE_INTERFACE) "; "  \
+    "setenv prefix /boot/; " 						\
+    "sysboot ${devtype} ${devnum}:${distro_bootpart_hex} any ${scriptaddr} ${prefix}extlinux/extlinux.conf; " \
     "run mender_try_to_recover"
 
 #endif /* !MENDER_AUTO_PROBING */
-- 
2.7.4

